Tasks/VsTest/VSTest.ps1

```diff
@@ -52,11 +52,18 @@ if($testAssemblyFiles)
 
     $resultFiles = Find-Files -SearchPattern "*.trx" -RootFolder $testResultsDirectory 
 
-    Publish-TestResults -Context $distributedTaskContext -TestResultsFiles $resultFiles -TestRunner "VSTest" -Platform $platform -Configuration $configuration
+    if($resultFiles)
+    {
+        Publish-TestResults -Context $distributedTaskContext -TestResultsFiles $resultFiles -TestRunner "VSTest" -Platform $platform -Configuration $configuration
```

Comment:
Make sure you test one item in the resultFiles and multiple items in the resultFiles. We have seen some difficulties in passing lists to cmdlets.